apis/Google.Cloud.Firestore/Google.Cloud.Firestore/WatchStream.cs

```diff
@@ -206,10 +206,15 @@ async Task CloseStreamAsync()
             }
 
             // Create a new enumerator for the retry attempt sequence, starting with a backoff of zero.
-            IEnumerator<RetryAttempt> CreateRetryAttemptSequence() =>
-                RetryAttempt
+            IEnumerator<RetryAttempt> CreateRetryAttemptSequence()
+            {
+                var iterator = RetryAttempt
                     .CreateRetrySequence(_backoffSettings, _scheduler, initialBackoffOverride: TimeSpan.Zero)
                     .GetEnumerator();
+                // Make sure Current is already valid
+                iterator.MoveNext();
```

Comment:
I'm a bit surprised this is required. What's consuming this that assumes the enumerator has already had `.MoveNext()` called?

Reply:
All of the rest of the code :) We do a MoveNext *after* retrying. I could move that instead, but I think I'll keep it here at least for now.